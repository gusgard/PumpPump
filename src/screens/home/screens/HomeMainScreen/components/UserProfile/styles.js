import { StyleSheet } from 'react-native'

import { horizontal, vertical, fontSize } from '@themes'

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingHorizontal: horizontal.small,
    paddingTop: vertical.small,
    paddingBottom: vertical.xxSmall,
  },
  texts: {
    flex: 1,
    marginLeft: horizontal.small,
  },
  name: {
    fontSize: fontSize.large,
    fontWeight: 'bold',
    marginBottom: vertical.xxSmall,
  },
})
