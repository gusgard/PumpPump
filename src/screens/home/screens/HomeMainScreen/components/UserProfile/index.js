import React, { PureComponent } from 'react'
import { View, Text } from 'react-native'
import PropTypes from 'prop-types'

import { Thumbnail } from '@components'
import Description from '../Description'

import styles from './styles'

export default class UserProfile extends PureComponent {
  static propTypes = {
    name: PropTypes.string.isRequired,
    bio: PropTypes.array.isRequired,
    picture: PropTypes.object.isRequired,
  };

  render() {
    const { name, bio, picture } = this.props
    return (
      <View style={styles.container}>
        <Thumbnail source={picture} />
        <View style={styles.texts}>
          {/* NAME */}
          <Text style={styles.name}>{name}</Text>
          {/* DESCRIPTION */}
          <Description bio={bio} />
        </View>
      </View>
    )
  }
}
