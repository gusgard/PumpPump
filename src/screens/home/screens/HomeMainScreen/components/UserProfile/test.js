import { shallow } from 'enzyme';
import React from 'react';

import UserProfile from './';

const user = {
  description: [],
  name: '',
  thumbnail: {},
};

it('renders correctly', () => {
  const wrapper = shallow(
    <UserProfile
      name={user.name}
      description={user.description}
      picture={user.thumbnail}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});
