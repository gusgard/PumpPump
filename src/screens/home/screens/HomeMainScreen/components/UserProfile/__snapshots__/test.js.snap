// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingBottom": 16.675,
      "paddingHorizontal": 28.125,
      "paddingTop": 50.025,
    }
  }
>
  <Thumbnail
    size={100}
    source={Object {}}
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 28.125,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 16.675,
        }
      }
    />
    <Description
      texts={Array []}
    />
  </View>
</View>
`;
